### ISSUE TITLE
Icons show question mark and no colour

### ISSUE THREAD

**Issue by B0ndo2** [maxwroc/battery-state-card#657]
Icons show question mark and no colour

**Expected behavior**
The icons should have colours based on battery level

**YAML configuration**
```yaml

type: custom:auto-entities
card:
  type: custom:battery-state-card
  title: Batteries
  sort_by_level: asc
  collapse: 5
  tap_action: more-info
  color_thresholds:
    - value: 10
      color: '#ff0000'
    - value: 50
      color: '#fa9e0a'
    - value: 100
      color: '#00ff00'
filter:
  include:
    - entity_id: sensor.*_battery
sort:
  method: state
  numeric: false
  ignore_case: true
  reverse: true
  ip: false


```
![image](https://github.com/maxwroc/battery-state-card/assets/75389943/8446914a-f876-47b6-8e1d-c48d0f09c3ca)

**Version**
3.1.5

**Comment by maxwroc** [comment]
What integration gives you these entities?

Can you please paste the entity and its attributes (screenshot) from the Home Assistant Developer Tools?

I think that the reason why the battery level was not correctly recognized is that entity state contains the "%" which is not prefixed with space

**Comment by maxwroc** [comment]
Can you please add to your config the following:
```yaml
default_state_formatting: false
```

Please tell me if this helps and if so what is your locale settings in HA

**Comment by B0ndo2** [comment]
That removed the question mark but all the icons now are green regardless of the threshold (I changed the 50% threshold to 60% to get a different colour)

![image](https://github.com/maxwroc/battery-state-card/assets/75389943/33070a69-a2dc-4708-936a-1c0e3bbfc4d3)

**Comment by papperone** [comment]
I have some problem since last version, it was working with all colors according to value, now only battery icon with question mark
![immagine](https://github.com/maxwroc/battery-state-card/assets/28104633/04598391-c0f5-4710-a6a0-306ede046ac2)

**Comment by B0ndo2** [comment]
> I have some problem since last version, it was working with all colors according to value, now only battery icon with question mark ![immagine](https://private-user-images.githubusercontent.com/28104633/297780925-04598391-c0f5-4710-a6a0-306ede046ac2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU1OTAyMDgsIm5iZiI6MTcwNTU4OTkwOCwicGF0aCI6Ii8yODEwNDYzMy8yOTc3ODA5MjUtMDQ1OTgzOTEtYzBmNS00NzEwLWE2YTAtMzA2ZWRlMDQ2YWMyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE4VDE0NTgyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJkYzZlMDU2MGQxMzFmNGRkYjMxNDhkNmM0NmRiYmY4ZjdlN2ZiMDg2N2I3Nzg0NjM0ZDJhNDUxYTNjZDEzYmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.v0pOLoZEcfDiInyK4fsjZtCJC6LpvbhVMaw2mRjdO1A)

Adding this in the card config fixed for me for the question mark but the colours are still the same for all levels
default_state_formatting: false

**Comment by badnetmask** [comment]
The `default_state_formatting: false` worked for me as well. My locale config is US/English/US customary.

**Comment by tomlut** [comment]
Just noticed this as well. I'm on the February beta release though and thought it may be due to this:

https://developers.home-assistant.io/blog/2024/01/30/ha-state-icon-properties-changes/

**Comment by bdormael** [comment]
same issue here since beta 2024.2 (Feb release)

**Comment by midiwidi** [comment]
I had the same problem (icon with a question mark and no color). "default_state_formatting: false" fixed both problems for me. It also fixed an issue with sorting. Without "default_state_formatting: false" the order was 100%,100%,17%,40%,69%,94% and with it is correct 17%,40%,69%,94%,100%,100%.
Sounds to me as if the entity state (battery level) is interpreted as a string instead of a number and can therefore not be mapped to the right symbol and color or sorted correctly..

**Comment by tomlut** [comment]
Yeah it fixed it for me too.

**Comment by maxwroc** [comment]
@midiwidi yes this is the exact problem. I have assumed that the formatted state which comes from HA will always be in the format of `[value] [unit]` but it looks like the space is gone in many cases after 2024.2 release.

@tomlut the change to ha-state-icon should not have any impact as this card uses ha-icon element only

**Comment by maxwroc** [comment]
Should be fixed now in v3.1.6
